### ISSUE TITLE
Suggest new course schedule

### ISSUE THREAD

**Issue by richelbilderbeek** [UPPMAX/programming_formalisms#45]
We had ideas for a new order of discussing course topics, most notably by getting the workflow in place in day 1, so that we can focus on the meat of the course from then. 

What about those interested suggest a new order?

**Comment by richelbilderbeek** [comment]
I am interested in doing this: yes

Assign the others so they can share if they are interested yes/no

**Comment by LarsEklund** [comment]
yes, we could do class discovery through list and then show how to refine  a class list through the use of diagrams to find missning classes and design elements. and what generates a combination of classes and so on.

**Comment by bclaremar** [comment]
Yes

**Comment by richelbilderbeek** [comment]
My suggested order can be found at <https://github.com/UPPMAX/programming_formalisms/tree/main/lesson_plans/2024_autumn#suggested-new-schedule>

**Comment by richelbilderbeek** [comment]
At the meeting of 2024-09-11 we got to:

- **Monday morning**: 
  - Björn: course overview
  - Björn: basic git
- **Monday afternoon**: Requirements modeling and risk assessment
    - Lars: Define the project
    - Make formal requirements model
    - Make formal risk assessment
- **Tuesday**: Git setup and version control, from a reproducible resarch and social coding/development perspective
    - Organize the project 
- **Wednesday**: Test-Driven Development (TDD) and function design
- **Thursday and Friday morning**: Test-Driven Development and class design (TDD) 
    - Implement the project as a package
- **Friday afternoon**: Deployment and documentation

**Comment by richelbilderbeek** [comment]
At [https://github.com/UPPMAX/programming_formalisms/blob/main/meeting_notes/20241108.md](https://github.com/UPPMAX/programming_formalisms/blob/main/meeting_notes/20241108.md) we decided to freeze the schedule. 

Done!
